better ETC for finished workflows

camallen opened the pull request:
Fixes https://www.zooniverse.org/talk/17/331857?comment=553695

don't calc and eta for complete workflows, just report 0. 

This looks like i should be extracting the ETA to a class as i've got a props data clump here. Thoughts on best practice for this instead of just 2 functions to create the html. 

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] Did you deploy a staging branch?
- [ ] Did you get any type checking errors from [Babel Typecheck](https://github.com/codemix/babel-plugin-typecheck)


## Optional

- [ ] If it's a new component, is it in ES6? Is it clear of warnings from ESLint?
- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?

eatyourgreens commented:
What @srallen said. Calculate the ETA in the parent then pass the result into the ETA component as a prop.

camallen commented:
updated - i've moved the calculation up to the parent component (thanks @srallen and @eatyourgreens), leaving the functional component as is. OOI should we report 0 or just nothing?

eatyourgreens commented on app/pages/project/stats/stats.jsx:
```diff
@@ -177,46 +177,20 @@ GraphSelect.propTypes = {
- }
-
-export const Eta = (props) => {
+export const Eta = (eta) => {
```
Small syntax comment. `eta` here should be named `props`.

eatyourgreens commented on app/pages/project/stats/stats.jsx:
```diff
@@ -177,46 +177,20 @@ GraphSelect.propTypes = {
     output = (
       <div>
-        <span className="progress-stats-label">ETC*</span> {`${eta} days`}
+        <span className="progress-stats-label">ETC*</span> {`${eta.numDays} days`}
```
`props.numDays` instead of `eta.numDays`.

camallen commented:
props are indeed passed to components! updated.
